napari/_qt/qt_viewer.py

```diff
@@ -328,6 +328,27 @@ def __init__(self, viewer: ViewerModel, show_welcome_screen: bool = False):
         # bind shortcuts stored in settings last.
         self._bind_shortcuts()
 
+        # set dask preferences from settings.
+
+        settings = get_settings()
+
+        self._update_dask_settings()
+
+        settings.application.events.dask.connect(self._update_dask_settings)
+
+    def _update_dask_settings(self, event=None):
```

Comment:
I don't like that this function takes no parameters, but rather grabs a global value when triggered. Could this be engineered so that the event contains the specific values, and the connected function takes those values as parameters? If I remember correctly, event values are now unpacked for callbacks? ie the dask configuration dictionary is either passed directly as an argument to the callback, or it's passed in `event.value`, I can't remember. (It's easy to check by using `.connect(print)`.)

Then this function could be something like:

```python
enabled = setting['enabled']
size = setting['cache']
resize_dask_cache(int(enabled) * size * 1e6)
```

Reply:
I made a change here in order to use the event.value in the callback.  I tried to pass in a dictionary only by doing 
```
settings.application.events.dask.connect(lambda x: self._update_dask_settings(x.value))
```
but the tests were failing because something wasn't being cleaned up properly.  So, this function now accepts either an event that must have a dictionary in event.value or it will accept a dictionary.   I'm not sure if there is a better way?